iac-extensions/spring-config/src/test/java/org/sonar/iac/springconfig/parser/properties/PropertiesParserBaseVisitorTest.java

```diff
@@ -1026,14 +1026,73 @@ void shouldVerifyTextRangesWhenCR() {
   }
 
   @Test
-  void shouldThrowExceptionForEmptyKey() {
+  void shouldParseEmptyKey() {
     var code = "=abc";
 
-    var exception = catchException(() -> parseProperties(code));
+    parseProperties(code);
+
+    assertThat(visitor.visited()).containsExactly(
+      "visitPropertiesFile =abc<EOF><EOF>",
+      "visitRow =abc<EOF>",
+      "visitInvalidLine =abc<EOF>",
+      "visitKey abc",
+      "visitEol <EOF>");
+  }
+
+  @Test
+  void shouldParseEmptyKeyBetweenOtherKeys() {
+    var code = """
+      foo=bar
+      =abc
+      foo2=bar2""";
+
+    parseProperties(code);
+
+    assertThat(visitor.visited()).containsExactly(
+      "visitPropertiesFile foo=bar\\n=abc\\nfoo2=bar2<EOF><EOF>",
+      "visitRow foo=bar\\n",
+      "visitLine foo=bar\\n",
+      "visitKey foo",
+      "visitKey bar",
+      "visitEol \\n",
+      "visitRow =abc\\n",
+      "visitInvalidLine =abc\\n",
+      "visitKey abc",
+      "visitEol \\n",
+      "visitRow foo2=bar2<EOF>",
+      "visitLine foo2=bar2<EOF>",
+      "visitKey foo2",
+      "visitKey bar2",
+      "visitEol <EOF>");
+  }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"=", ":"})
+  void shouldParseLineContainingDelimiterBetweenOtherKeys(String delimiter) {
+    var code = """
+      foo=bar
+      %sabc
```

Comment:
Maybe also add a test with both key and value missing, so just with a delimiter on a line?